src/sentry/models/project.py

```diff
@@ -475,13 +477,23 @@ def is_valid_platform(value):
             return True
         return integration_doc_exists(value)
 
+    @staticmethod
+    def outbox_for_update(project_identifier: int, organization_identifier: int) -> RegionOutbox:
+        return RegionOutbox(
+            shard_scope=OutboxScope.ORGANIZATION_SCOPE,
+            shard_identifier=organization_identifier,
+            category=OutboxCategory.PROJECT_UPDATE,
+            object_identifier=project_identifier,
+        )
+
     def delete(self, **kwargs):
         from sentry.models import NotificationSetting
 
         # There is no foreign key relationship so we have to manually cascade.
         NotificationSetting.objects.remove_for_project(self)
-
-        return super().delete(**kwargs)
+        with transaction.atomic(), in_test_psql_role_override("postgres"):
+            Project.outbox_for_update(self.id, self.organization_id).save()
```

Comment:
once we have the outbox processor for this would we be able to remove `NotificationSettings.objects.remove_for_project()` above? As that seems to be what the outbox message would do as well.

Reply:
Yes, that needs to change as well.  It gets a little tricky because the way NotificationSettings are setup, they don't have a direct foreign key but a composite key relationship.  It's doable, and I probably have to do it to split the silo, so thanks for pointing it out :P Next PR